packages/api/src/hooks/Composer.tsx

```diff
@@ -540,6 +540,12 @@ const ComposerCore = ({
     [scrollToEndButtonMiddleware]
   );
 
+  // For useFiles hook
+  const [files, setFiles] = useState<File[]>([]);
```

Comment:
Using `useState` here, instead of Redux, will complicate the code later.

In your latter code, sometimes, it call `useSendFiles`, and in other times, it call, `useAPISubmitSendBox`.

If the attachment list is maintained inside Redux, the latter code will not need to call `useSendFiles`.